File: frontend/src/types/review.ts
export interface ReviewScores {
  entry_timing: number
  exit_timing: number
  stop_loss: number
  take_profit: number
  position_sizing: number
  holding_period: number
  discipline: number
  risk_reward: number
}

export interface TradeReview {
  id: number
  ts_code: string
  stock_name: string
  plan_id: number | null
  total_buy_amount: number
  total_sell_amount: number
  total_fee: number
  realized_pnl: number
  trade_count: number
  first_trade_date: string
  last_trade_date: string
  holding_days: number
  scores: ReviewScores
  overall_score: number
  analysis: string
  improvement: string | null
  user_notes: string | null
  created_at: string
  updated_at: string
}

export interface ReviewStats {
  total_reviews: number
  avg_overall_score: number
  avg_scores: ReviewScores
  best_dimension: string
  worst_dimension: string
  win_count: number
  loss_count: number
  total_reviewed_pnl: number
}

export type PatternType = 'strength' | 'weakness'
export type PatternStatus = 'active' | 'resolved' | 'dismissed'

export interface BehaviorPattern {
  id: number
  pattern_type: PatternType
  title: string
  description: string
  dimension: string | null
  evidence_ids: number[]
  status: PatternStatus
  created_at: string
  updated_at: string
}

File: backend/app/models/review.py
from sqlalchemy import Column, Float, ForeignKey, Index, Integer, String, Text

from app.database import Base


class TradeReview(Base):
    __tablename__ = "trade_review"

    id = Column(Integer, primary_key=True, autoincrement=True)
    ts_code = Column(String(20), nullable=False, index=True)
    stock_name = Column(String(50), nullable=False)
    plan_id = Column(Integer, ForeignKey("trading_plan.id", ondelete="SET NULL"), nullable=True)
    total_buy_amount = Column(Float, nullable=False)
    total_sell_amount = Column(Float, nullable=False)
    total_fee = Column(Float, nullable=False)
    realized_pnl = Column(Float, nullable=False)
    trade_count = Column(Integer, nullable=False)
    first_trade_date = Column(String(10), nullable=False)
    last_trade_date = Column(String(10), nullable=False)
    holding_days = Column(Integer, nullable=False)
    scores = Column(Text, nullable=False)
    overall_score = Column(Float, nullable=False)
    analysis = Column(Text, nullable=False)
    improvement = Column(Text, nullable=True)
    user_notes = Column(Text, nullable=True)
    created_at = Column(String(30), nullable=False)
    updated_at = Column(String(30), nullable=False)

    __table_args__ = (
        Index("ix_review_ts_code", "ts_code"),
        Index("ix_review_plan_id", "plan_id"),
    )


class BehaviorPattern(Base):
    __tablename__ = "behavior_pattern"

    id = Column(Integer, primary_key=True, autoincrement=True)
    pattern_type = Column(String(20), nullable=False)
    title = Column(String(100), nullable=False)
    description = Column(Text, nullable=False)
    dimension = Column(String(30), nullable=True)
    evidence_ids = Column(Text, nullable=True)
    status = Column(String(20), nullable=False, default="active")
    created_at = Column(String(30), nullable=False)
    updated_at = Column(String(30), nullable=False)
